### PR TITLE
[Julius Gambe] iP

### PR DESCRIPTION
**Pull request by juliusgambe**
# Gobble Gobble

Initial Pull Request of **Gobble** **Gobble** 🏁.  This project is based on [Duke](https://dev.java/duke/).

## Changes
- Add Ui
- Add Storage
- Add Parser
- Add Command
- Add Task

## Type of Tasks
1. TodoTask
2. DeadlineTask
3. EventTask

## Type of Commands
1. AddCommand
2. DeleteCommand
3. ExitCommand
4. ListCommand
5. MarkCommand
6. UnmarkCommand

## Review Checklist
- [ ] My code follows the **style guidelines**
- [ ] My code follows **OOP**
- [ ] My code satisfies a _minimal_ amount of tests critical to the project
- [ ] My code has **javadocs** as its documentation

## Code Starting Point
Code starts at `Duke.java`

```java
 public void run() {
        ui.showWelcomeMessage();
        Scanner scanner = new Scanner(System.in);
        while (true) {
            String userInput = scanner.nextLine().trim();
            try {
                Command command = Parser.parse(userInput);
                command.execute(tasks, ui, storage);
                if (command.isExit()) {
                    break;
                }

            } catch (DukeException e) {
                ui.showError(e.getMessage());
            }

        }
        scanner.close();
    }
```

### CODE REVIEW

File: src/main/java/duke/enums/TaskType.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package duke.enums;
+
+/**
+ * Represents a TaskType enum that contains the types of tasks.
+ */
+public enum TaskType {
+    TODO,
+    DEADLINE,
+    EVENT
+}
```
Comment (antonTan96) [inline src/main/java/duke/enums/TaskType.java:10]: I like how you seperated the task types into another file :D

File: src/main/java/duke/task/DeadlineTask.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ */
+public class DeadlineTask extends Task {
+    private final LocalDate by;
+
+    /**
+     * Constructor for DeadlineTask.
+     *
+     * @param description of the task.
+     * @param by          deadline of the task.
+     */
```
Comment (antonTan96) [inline src/main/java/duke/task/DeadlineTask.java]: I'm not sure if this follows the coding standard.
I would much prefer `@param description The description of the task`

File: src/main/java/duke/enums/TaskType.java (hunk 1 above)
Comment (antonTan96) [inline src/main/java/duke/enums/TaskType.java:10]: I like how you seperated the task types into another file :D

**Review by antonTan96 — COMMENTED** [review]
Looks good to me! No glaring issues and the coding standard is followed.

File: src/main/java/duke/ui/GobbleMessage.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+import javafx.geometry.Pos;
+import javafx.scene.Node;
+import javafx.scene.control.Label;
+import javafx.scene.layout.HBox;
+import javafx.scene.layout.Region;
+
+/**
+ * Controller for a Message Bubble in the GUI.
+ */
+public class GobbleMessage extends HBox {
```
Comment (jianyangg) [inline src/main/java/duke/ui/GobbleMessage.java:22]: Wow! What an interesting and creative class name. 🤯

File: src/main/java/duke/command/UnmarkCommand.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+     * Unmarks a task as done and displays it as undone.
+     *
+     * @param taskList list of tasks
+     * @param storage  storage
+     */
+    @Override
+    public GobbleMessage execute(TaskList taskList, Storage storage) {
+        taskList.unmarkTask(index);
+        storage.saveListToDisk(taskList.getTasks());
+        return GobbleMessage.getDukeDialog("Nice! I've unmarked this task as undone:\n" + taskList.getTask(index),
```
Comment (jianyangg) [inline src/main/java/duke/command/UnmarkCommand.java]: Could this GobbleMessage be gobbling up too many characters (i.e., past the limit of 120 chars)?

File: src/main/java/duke/ui/MainWindow.java [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+            GobbleMessage error = GobbleMessage.getDukeDialog(e.getMessage(), "Error");
+            this.addMessages(userMessage, error);
+        }
+        userInput.clear();
+    }
+
+    /**
+     * Adds messages to the dialog container.
+     *
+     * @param messages messages to be added.
```
Comment (jianyangg) [inline src/main/java/duke/ui/MainWindow.java]: Similar to what Anton mentioned earlier and referring to CS2103T Java Conventions, we'll have to capitalise our descriptive text after the params. See this snippet from the textbook: 

`@param zone Zone of position.`

**Review by jianyangg — COMMENTED** [review]
LGTM! Just a few nits to fix such as JavaDocs conventions and line character limit. Very creative work and elegant code nonetheless!

**Pull request closed without merging**